app/src/main/java/com/vitorpamplona/amethyst/ui/screen/ChatroomListFeedView.kt

```diff
@@ -73,18 +88,53 @@ fun ChatroomListFeedView(viewModel: FeedViewModel, accountViewModel: AccountView
 private fun FeedLoaded(
     state: FeedState.Loaded,
     accountViewModel: AccountViewModel,
-    navController: NavController
+    navController: NavController,
+    markAsRead: MutableState<Boolean>,
 ) {
     val listState = rememberLazyListState()
 
+    val accountState by accountViewModel.accountLiveData.observeAsState()
+    val account = accountState?.account ?: return
+    val notificationCacheState = NotificationCache.live.observeAsState()
+    val notificationCache = notificationCacheState.value ?: return
+    val context = LocalContext.current.applicationContext
+
+    if (markAsRead.value) {
+        LaunchedEffect(key1 = notificationCache) {
+            for (note in state.feed.value) {
+                note.event?.let {
+                    var route = ""
+
+                    if (note.channel != null) {
+                        route = "Channel/${note.channel!!.idHex}"
+                    } else {
+                        val replyAuthorBase = note.mentions?.first()
+                        var userToComposeOn = note.author!!
+                        if (replyAuthorBase != null) {
+                            if (note.author == account.userProfile()) {
+                                userToComposeOn = replyAuthorBase
+                            }
+                        }
+                        route = "Room/${userToComposeOn.pubkeyHex}"
+                    }
```

Comment:
This code is duplicated from `ChatroomCompose`, but at the moment there is no obvious place to extract it for reuse.